pkg/controller/ipam/validate.go

```diff
@@ -107,6 +145,60 @@ func humanReadableIPRanges(ipRanges []crdv1alpha2.SubnetIPRange) string {
 	return fmt.Sprintf("[%s]", strings.Join(strs, ","))
 }
 
+func rangesOverlap(r1, r2 crdv1alpha2.SubnetIPRange) bool {
+	if r1.CIDR == "" {
+		if r2.CIDR == "" {
+			r1start := net.ParseIP(r1.Start).To16()
+			r1end := net.ParseIP(r1.End).To16()
+			r2start := net.ParseIP(r2.Start).To16()
+			r2end := net.ParseIP(r2.End).To16()
+
+			if bytes.Compare(r2start, r1start) >= 0 && bytes.Compare(r2start, r1end) <= 0 {
+				return true
+			} else if bytes.Compare(r1start, r2start) >= 0 && bytes.Compare(r1start, r2end) <= 0 {
+				return true
+			}
+		} else {
+			_, cidr2, _ := net.ParseCIDR(r2.CIDR)
+			return cidr2.Contains(net.ParseIP(r1.Start)) || cidr2.Contains(net.ParseIP(r1.End))
+		}
+	} else {
+		_, cidr1, _ := net.ParseCIDR(r1.CIDR)
+		if r2.CIDR == "" {
+			return cidr1.Contains(net.ParseIP(r2.Start)) || cidr1.Contains(net.ParseIP(r2.End))
+		} else {
+			_, cidr2, _ := net.ParseCIDR(r2.CIDR)
+			return cidr1.Contains(cidr2.IP) || cidr2.Contains(cidr1.IP)
+		}
+	}
+	return false
+}
+
+func validateIPRange(r crdv1alpha2.SubnetIPRange) (bool, string) {
+	// Validate the integrity of IPs within the IP range
+	gateway := net.ParseIP(r.Gateway)
+	if r.CIDR != "" {
+		_, cidr, _ := net.ParseCIDR(r.CIDR)
+		if !cidr.Contains(gateway) {
+			return false, fmt.Sprintf(
+				"Range is invalid. Gateway %s is unreachable from CIDR %s", r.Gateway, r.CIDR)
```

Comment:
In my understanding, as the feature requires the Pod CIDR is in the same subnet of Node IP addresses. The CIDR will be a subset of the underlay subnet and the gateway will not be within the CIDR. For example, the underlay subnet is 192.168.0.0/22, the Node CIDR is 192.168.0.0/24, the Pod CIDR is 192.168.1.0/24, but the gateway 192.168.0.1 is not within 192.168.1.0/24. 
I think it should check PodCIDR/r.PrefixLength contains the gateway IP.
@GraysonWu @annakhm could you confirm?

Reply:
I should @gran-vmv, not Grayson.

Reply:
@annakhm Do we need to check it here: https://github.com/antrea-io/antrea/blob/e0e2e07d91cf89055960d3d8f226b703554ded8b/pkg/ipam/poolallocator/allocator.go#L73

Reply:
@gran-vmv is my understanding correct? We should check whether PodCIDR/PrefixLength contains the gateway IP, instead of the provided PodCIDR itself, right?

Reply:
I believe we should check that subnet CIDR defined by `SubnetInfo.Gateway`/`SubnetInfo.PrefixLength` should contain the `CIDR`/range that we are validating.